### PR TITLE
ensure gcp container cluster is not public rule

### PR DESCRIPTION
**Pull request by doronafota**

### CODE REVIEW

File: cloudrail/knowledge/rules/gcp/non_context_aware/container_cluster_is_not_public_rule.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+class ContainerClusterIsNotPublictRule(GcpBaseRule):
+    def get_id(self) -> str:
+        return 'non_car_gke_control_plane_ensure_not_public'
+
+    def execute(self, env_context: GcpEnvironmentContext, parameters: Dict[ParameterType, any]) -> List[Issue]:
+        issues: List[Issue] = []
+        for container_cluster in env_context.container_cluster:
+            should_alert = True
+            if container_cluster.master_authorized_networks_config:
```
Comment (Imanuel-Mizrahi) [inline cloudrail/knowledge/rules/gcp/non_context_aware/container_cluster_is_not_public_rule.py:17]: I think that this condition might be simplified.
Something like:
if not container_cluster.master_authorized_networks_config or not any(cidr_obj.cidr_block for cidr_obj in container_cluster.master_authorized_networks_config.cidr_blocks):
And remove the "should_alert" variable completely.

**Review by Imanuel-Mizrahi — APPROVED** [review]

File: cloudrail/knowledge/rules/gcp/non_context_aware/container_cluster_is_not_public_rule.py (hunk 1 above)
Comment (Tomerva) [reply cloudrail/knowledge/rules/gcp/non_context_aware/container_cluster_is_not_public_rule.py:17]: i partially agree - i just dont understand why you took the negative route
you could do this:
`if container_cluster.master_authorized_networks_config and any(cidr_obj.cidr_block for cidr_obj in container_cluster.master_authorized_networks_config.cidr_blocks): issue.append(...)`

on a side note, you should have used `break` after `should_alert = False`

File: tests/knowledge/rules/gcp/non_context_aware/test_container_cluster_is_not_public_rule.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        ]
+    )
+
+    def test_container_cluster_is_not_public(self, unused_name: str, cidr_blocks: dict, total_issues: int, rule_status: RuleResultType):
+        # Arrange
+        container_clusters: List[GcpContainerCluster] = []
+        for i in range(2):
+            container_clusters.append(create_empty_entity(GcpContainerCluster))
+
+        for i, cidr_block in enumerate(cidr_blocks):
```
Comment (Tomerva) [inline tests/knowledge/rules/gcp/non_context_aware/test_container_cluster_is_not_public_rule.py:29]: you should merge the two loops into one

**Review by Tomerva — CHANGES_REQUESTED** [review]

File: cloudrail/knowledge/rules/gcp/non_context_aware/container_cluster_is_not_public_rule.py (hunk 1 above)
Comment (doronafota) [reply cloudrail/knowledge/rules/gcp/non_context_aware/container_cluster_is_not_public_rule.py:17]: @Imanuel-Mizrahi @Tomerva 
you are right - i changed it to 
```if not container_cluster.master_authorized_networks_config or 
          any(not cidr_obj.cidr_block for cidr_obj in container_cluster.master_authorized_networks_config.cidr_blocks):
```

File: tests/knowledge/rules/gcp/non_context_aware/test_container_cluster_is_not_public_rule.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    @parameterized.expand(
+        [
+            ["One cluster is public", ["10.0.0.0/8", ""], 1, RuleResultType.FAILED],
+            ["Both clusters are public", ["", ""], 2, RuleResultType.FAILED],
+            ["Both clusters are private", ["10.0.0.0/8", "10.0.0.0/8"], 0, RuleResultType.SUCCESS],
+        ]
+    )
+
+    def test_container_cluster_is_not_public(self, unused_name: str, cidr_blocks: dict, total_issues: int, rule_status: RuleResultType):
```
Comment (Tomerva) [inline tests/knowledge/rules/gcp/non_context_aware/test_container_cluster_is_not_public_rule.py:23]: cidr_blocks is a `List[str]`

**Review by Tomerva — APPROVED** [review]

**Pull request merged**